pkgs/applications/science/math/R/default.nix

```diff
@@ -30,7 +30,19 @@ stdenv.mkDerivation rec {
 
   patches = [
     ./no-usr-local-search-paths.patch
-    ./fix-failing-test.patch
+  ] ++ lib.optionals (!withRecommendedPackages) [
+    (fetchpatch {
+       name = "fix-tests-without-recommended-packages.patch";
+       url = "https://github.com/wch/r-source/commit/7715c67cabe13bb15350cba1a78591bbb76c7bac.patch";
+       # this part of the patch reverts something that was committed after R 4.1.0, so ignore it.
+       excludes = [ "tests/Pkgs/xDir/pkg/DESCRIPTION" ];
+       sha256 = "sha256-iguLndCIuKuCxVCi/4NSu+9RzBx5JyeHx3K6IhpYshQ=";
+    })
+    (fetchpatch {
+      name = "use-codetools-conditionally.patch";
+      url = "https://github.com/wch/r-source/commit/7543c28b931db386bb254e58995973493f88e30d.patch";
+      sha256 = "sha256-+yHXB5AItFyQjSxfogxk72DrSDGiBh7OiLYFxou6Xlk=";
+    })
```

Comment:
Those patches don't break anything when applied all the time, right?

Reply:
They are reducing test coverage, so I guess they will allow R to build all the time, but funny things might happen during runtime.

Reply:
Updating to 4.1.1 removes the need for these patches.